### PR TITLE
Fix checking the lock file too often while writing a backup

### PR DESCRIPTION
**Pull request by vitlibar**
### Changelog category:
- Bug Fix

### Changelog entry:
Fix checking the lock file too often while writing a backup

### CODE REVIEW

File: src/Backups/BackupImpl.cpp [hunk 1]
```diff
@@ -144,6 +144,7 @@ void BackupImpl::open(const ContextPtr & context)
         if (!uuid)
             uuid = UUIDHelpers::generateV4();
         lock_file_name = use_archive ? (archive_params.archive_name + ".lock") : ".lock";
+        lock_file_before_first_file_checked = false;
```
Comment (nikitamikhaylov) [inline src/Backups/BackupImpl.cpp:147]: Why can we do this just in constructor of BackupImpl? I afraid we can start to use  `BackupImpl::writeFile` when `open_mode != OpenMode::WRITE`

**Review by nikitamikhaylov — APPROVED** [review]

File: src/Backups/BackupImpl.cpp (hunk 1 above)
Comment (vitlibar) [reply src/Backups/BackupImpl.cpp:147]: Files can be written only in `OpenMode::WRITE`.

**Comment by tavplubix** [comment]
Btw, is it expected that the lock file suddenly disappears sometimes? https://github.com/ClickHouse/ClickHouse/issues/47729

**Comment by vitlibar** [comment]
> Btw, is it expected that the lock file suddenly disappears sometimes? See #47729 and #49506

It doesn't disappear, just a bad error message. I'll answer there.

**Pull request merged**